stellargraph/mapper/full_batch_generators.py

```diff
@@ -238,3 +238,209 @@ def flow(self, node_ids, targets=None):
+    to get an object that can be used as a Keras data generator.
+
+    This generator will supply the features array and the adjacency matrix to a
+    full-batch Keras graph ML model.  There is a choice to supply either a list of sparse
+    adjacency matrices (the default) or a list of dense adjacency matrices, with the `sparse`
+    argument.
+
+    For these algorithms the adjacency matrices require pre-processing and the default option is to
+    normalize each row of the adjacency matrix so that it sums to 1.
+    For customization a transformation (callable) can be passed that
+    operates on the node features and adjacency matrix.
+
+    Example::
+
+        G_generator = RelationalFullBatchNodeGenerator(G)
+        train_data_gen = G_generator.flow(node_ids, node_targets)
+
+        # Fetch the data from train_data_gen, and feed into a Keras model:
+        # Alternatively, use the generator itself with model.fit_generator:
+        model.fit_generator(train_gen, epochs=num_epochs, ...)
+
+    Args:
+        G (StellarGraphBase): a machine-learning StellarGraph-type graph
+        name (str): an optional name of the generator
+        transform (callable): an optional function to apply on features and adjacency matrix
+            the function takes (features, Aadj) as arguments.
+        sparse (bool): If True (default) a list of sparse adjacency matrices is used,
+            if False a list of dense adjacency matrices is used.
+
+    """
+
+    def __init__(self, G, name=None, sparse=True, transform=None):
+
+        if not isinstance(G, StellarGraphBase):
+            raise TypeError("Graph must be a StellarGraph object.")
+
+        self.graph = G
+        self.name = name
+        self.use_sparse = sparse
+
+        # Check if the graph has features
+        G.check_graph_for_ml()
+
+        # extract node, feature, and edge type info from G
+        self.node_list = list(G.nodes())
+
+        self.features = G.get_feature_for_nodes(self.node_list)
+
+        edge_types = sorted(set(e[-1] for e in G.edges))
+
+        node_index = dict(zip(self.node_list, range(len(self.node_list))))
+
+        # create a list of adjacency matrices - one adj matrix for each edge type
+        # an adjacency matrix is created for each edge type from all edges of that type
+        self.As = []
+
+        for edge_type in edge_types:
+
+            col_index = [
+                node_index[n1] for n1, n2, etype in G.edges if etype == edge_type
+            ]
+            row_index = [
+                node_index[n2] for n1, n2, etype in G.edges if etype == edge_type
+            ]
+            data = np.ones(len(col_index), np.float64)
+
+            # note that A is the transpose of the standard adjacency matrix
+            # this is to aggregate features from incoming nodes
+            A = sps.coo_matrix(
+                (data, (row_index, col_index)),
+                shape=(len(self.node_list), len(self.node_list)),
+            )
```

Comment:
It looks like the new `StellarGraph` class (under development) should have a method to create adjacency matrices for different relation types.  This should be captured somewhere 📣